tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/ErrorTest.java

```diff
@@ -76,15 +75,12 @@ public void testSelfReferencingObject() {
         BRunUtil.invoke(compileResult, "testSelfReferencingError");
     }
 
-    @Test
+    @Test(description = "Test runtime out of memory error", expectedExceptions = java.lang.RuntimeException.class,
+            expectedExceptionsMessageRegExp = "error: \\{ballerina}OutOfMemoryError \\{\"message\":\"Java heap " +
+                    "space\"}.*")
     public void testRuntimeOOMError() {
-        try {
-            CompileResult compileResult = BCompileUtil.compile("test-src/jvm/runtime-oom-error.bal");
-            BRunUtil.runMain(compileResult, new String[]{});
-        } catch (Throwable e) {
-            return;
-        }
-        Assert.fail("runtime out of memory errors are not handled");
+        CompileResult compileResult = BCompileUtil.compile("test-src/jvm/runtime-oom-error.bal");
+        BRunUtil.runMain(compileResult, new String[]{});
```

Comment:
we can get rid of the array because the method expects a vararg:
```suggestion
        BRunUtil.runMain(compileResult);
```

Reply:
I will look into this.